osu.Game/Screens/Play/HUD/BPMCounter.cs

```diff
@@ -38,7 +41,7 @@ protected override void Update()
             base.Update();
 
             //We don't want it going to 0 when we pause. so we block the updates
-            if (gameplayClock.IsPaused.Value) return;
+            if (gameplayClock.IsPaused.Value || player.GameplayState.HasFailed) return;
 
             // We want to check Rate every update to cover windup/down
             Current.Value = beatmap.Value.Beatmap.ControlPointInfo.TimingPointAt(gameplayClock.CurrentTime).BPM * gameplayClock.Rate;
```

Comment:
I find the usage of `gameplayClock.Rate` here very suspicious in general. Reasoning: Go into a replay, and fiddle with the playback speed slider. Should the bpm counter really change value depending on that in such a case?

Likely should be using [this extension method](https://github.com/ppy/osu/blob/9aab502adcc6e61bb675eafcfaf5368f828e1ab8/osu.Game/Screens/Play/GameplayClockExtensions.cs#L14) instead. At which point none of the magic guards above should be necessary any longer.